src/main/java/nl/utwente/atelier/pmd/AtelierPMDRenderer.java

```diff
@@ -210,8 +197,92 @@ public void end() throws IOException {
 
         this.getWriter().close();
     }
+    
+    private List<JsonObject> mGetSummaryMessage(Map<String, Integer> mRuleViolationStatistics) {
+        List<JsonObject> liResult = new ArrayList<JsonObject>();
+
+        JsonObject oPrivateMessage = new JsonObject();
+        oPrivateMessage.addProperty("submissionID", submissionID);
+        oPrivateMessage.addProperty("visibility", "private");
+        oPrivateMessage.addProperty("automated", true);
+
+        JsonObject oPublicMessage = new JsonObject();
+        oPublicMessage.addProperty("submissionID", submissionID);
+        oPublicMessage.addProperty("visibility", "public");
+        oPublicMessage.addProperty("automated", true);
+
+        StringBuilder sbPrivateMessage = new StringBuilder();
+        sbPrivateMessage.append("ZITA Summary:\n");
+
+        StringBuilder sbPublicMessage = new StringBuilder();
+        float iViolationScore = 0;
+        boolean[] arViolationFlags = new boolean[] { // Probably better to use a bit mask maybe.
```

Comment:
I think a bitmask would be even more unreadable. Maybe it makes more sense to just use three variables for this, as there isn't really any benefit to them being in a list, it just makes things more confusing. Even if you want to add more later, the list doesn't really makes things easier. I can't think of a real good way to solve this though (especially in Java).